### PR TITLE
Feature/component testing nw

### PR DESCRIPTION
**Pull request by nware1066**
### Description

This PR adds unit testing for the BookCard, BookContainer, Search, ToRead, HaveRead and GoalForm Components,
testing is not complete, as there is no integration testing and no sad path testing.

### Types of changes:
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Styling- no new features

### Has This Been Tested?
- [ ] No
- [ ] Yes

### What are the relevant tickets?

#14 

### Checklist:
- [ ] My code follows the code style of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/Components/GoalForm/GoalForm.test.js [hunk 1]
```diff
@@ -1,10 +1,40 @@
...
+  //     <Provider store={store}>
+  //       <GoalForm onClick={mockSetUserGoal}/>
+  //     </ Provider>
+  //   )
+  //
+  //   userEvent.type(screen.getByPlaceholderText('Add reading goal'), 5)
+  //
+  //   userEvent.click(screen.getByText('Submit'))
+  //   expect(mockSetUserGoal).toHaveBeenCalled()
+  // })
```
Comment (stacyp2006) [inline src/Components/GoalForm/GoalForm.test.js:39]: We could add a test to verify that the goal value updates.

File: src/Components/Search/Search.test.js [hunk 2]
```diff
@@ -2,9 +2,26 @@ import React from 'react';
...
+      </Provider>
+    )
+
+    const dropDown = screen.getByRole('combobox')
+    expect(screen.getByText('Choose a List to Browse')).toBeInTheDocument()
+    expect(screen.getByRole('button', {name: 'Submit'})).toBeInTheDocument()
+    expect(screen.getByRole('combobox')).toBeInTheDocument()
+    expect(dropDown).toHaveLength(5)
   })
 })
```
Comment (stacyp2006) [inline src/Components/Search/Search.test.js:27]: Again, we could add a test to verify the value updates.

**Review by stacyp2006 — APPROVED** [review]
I think this a great start to testing.

File: src/Components/BookCard/BookCard.test.js [hunk 3]
```diff
@@ -1,10 +1,61 @@
...
+import userEvent from '@testing-library/user-event';
 import '@testing-library/jest-dom';
+import { combineReducers, createStore } from 'redux';
+import { Provider } from 'react-redux';
+import allReducers from '../../reducers/index.js'
 
 describe('BookCard', () => {
-  it('Should work', () => {
-    expect(true).toBe(true);
+
```
Comment (aemcdonald) [inline src/Components/BookCard/BookCard.test.js:11]: Great unit test here, I like that you mock that functions were called when the buttons are clicked

File: src/Components/HaveRead/HaveRead.test.js [hunk 4]
```diff
@@ -2,9 +2,20 @@ import React from 'react';
 import HaveRead from './HaveRead';
 import { render, screen } from '@testing-library/react';
 import '@testing-library/jest-dom';
+import { combineReducers, createStore } from 'redux';
```
Comment (aemcdonald) [inline src/Components/HaveRead/HaveRead.test.js:5]: Great start here with a unit test. We will have to look into testing the overall integrated functionality with Redux

File: src/Components/Search/Search.test.js [hunk 5]
```diff
@@ -2,9 +2,26 @@ import React from 'react';
...
+
+    const store = createStore(allReducers);
+
+    render(
+      <Provider store={store}>
+        <Search />
+      </Provider>
+    )
+
+    const dropDown = screen.getByRole('combobox')
```
Comment (aemcdonald) [inline src/Components/Search/Search.test.js:21]: Awesome job figuring out how to test the dropdown & that it has the correct length!

**Review by aemcdonald — APPROVED** [review]
Great job getting us started on some initial unit tests.

**Pull request merged**